File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_22_template/java/_02_after/Multiply.java
package com.example._03_behavioral_patterns._22_template.java._02_after;

public class Multiply extends FileProcessor {

    public Multiply(String path) {
        super(path);
    }

    @Override
    protected int getResult(int result, int number) {
        return result *= number;
    }
}

File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_22_template/kt/_02_after/Multiply.kt
package com.example._03_behavioral_patterns._22_template.kt._02_after

class Multiply(
    path: String
) : FileProcessor(path) {

    override fun getResult(result: Int, number: Int): Int {
        var result = result
        return number.let { result *= it; result }
    }
}
